t/12-bmwqemu.t

```diff
@@ -0,0 +1,117 @@
+
+use 5.018;
+use warnings;
+use Test::More;
+use File::Temp 'tempdir';
+use File::Path 'make_path';
+use JSON;
+
+BEGIN {
+    unshift @INC, '..';
+}
+
+subtest 'test CASEDIR, PRJDIR and DISTRI handling' => sub {
+    my $dir = tempdir(TMPDIR => 1, CLEANUP => 1);
+    make_path("$dir/tests/test");
+
+    open(my $varsfh, '>', 'vars.json') || BAIL_OUT('can not create vars.json');
+    my $json = JSON->new->pretty->canonical;
+    print $varsfh $json->encode({DISTRI => 'test', PRJDIR => $dir});
+    close($varsfh);
+
+    eval {
+        use bmwqemu ();
+        $bmwqemu::scriptdir = $dir;
+        bmwqemu::init;
+    };
+    ok(!$@, 'init successful');
+
+    local $/;
+    open($varsfh, '<', 'vars.json') || BAIL_OUT('can not open vars.json for reading');
+    my $ret;
+    eval { $ret = JSON->new->relaxed->decode(<$varsfh>); };
+    die "parse error in vars.json:\n$@" if $@;
+    close($varsfh);
+    my %vars = %{$ret};
+    is($vars{DISTRI},  'test',            'DISTRI unchanged by init call');
+    is($vars{CASEDIR}, "$dir/tests/test", 'CASEDIR correctly determined');
+    is($vars{PRJDIR},  $dir,              'PRJDIR unchanged');
+};
+
+subtest 'test PRJDIR default' => sub {
+    my $dir = '/var/lib/openqa/share/tests/test';
+
+    open(my $varsfh, '>', 'vars.json') || BAIL_OUT('can not create vars.json');
+    my $json = JSON->new->pretty->canonical;
+    print $varsfh $json->encode({CASEDIR => $dir});
+    close($varsfh);
+
+    eval {
+        use bmwqemu ();
+        $bmwqemu::scriptdir = $dir;
+        bmwqemu::init;
+    };
+    ok(!$@, 'init successful');
+
+    local $/;
+    open($varsfh, '<', 'vars.json') || BAIL_OUT('can not open vars.json for reading');
+    my $ret;
+    eval { $ret = JSON->new->relaxed->decode(<$varsfh>); };
+    die "parse error in vars.json:\n$@" if $@;
+    close($varsfh);
+    my %vars = %{$ret};
+    ok(!$vars{DISTRI}, 'DISTRI not supplied and not set');
+    is($vars{CASEDIR}, $dir, 'CASEDIR unchanged');
+    is($vars{PRJDIR}, '/var/lib/openqa/share', 'PRJDIR set to default');
+};
+
+subtest 'test CASEDIR not under PRJDIR default' => sub {
+    my $dir = '/tmp/some/dir/tests/test';
+
+    open(my $varsfh, '>', 'vars.json') || BAIL_OUT('can not create vars.json');
+    my $json = JSON->new->pretty->canonical;
+    print $varsfh $json->encode({CASEDIR => $dir});
+    close($varsfh);
+
+    eval {
+        use bmwqemu ();
+        $bmwqemu::scriptdir = $dir;
+        bmwqemu::init;
+    };
+    ok($@, 'bmwqemu init failed');
+
+    local $/;
+    open($varsfh, '<', 'vars.json') || BAIL_OUT('can not open vars.json for reading');
+    my $ret;
+    eval { $ret = JSON->new->relaxed->decode(<$varsfh>); };
+    die "parse error in vars.json:\n$@" if $@;
+    close($varsfh);
+    my %vars = %{$ret};
+    ok(!$vars{DISTRI}, 'DISTRI not supplied and not set');
+    is($vars{CASEDIR}, $dir, 'CASEDIR unchanged');
+    ok(!$vars{PRJDIR}, 'PRJDIR not supplied and not set');
+};
```

Comment:
that test has a lot of duplication. Makes hard to see what are the differences among the three subtests. Can you extra helper function(s)?